### ISSUE TITLE
Error when upload to headset after rename

### ISSUE THREAD

**Issue by Iorkca** [Sombody101/APKognito#7]
Log pack attached.

Error code
[ERROR]   ! The following APKs failed to be renamed with their error reason:
com.MightyCoconut.WalkaboutMiniGolf.apk: adb: failed to stat downloads\\com.apkognito.WalkaboutMiniGolf\_2024-12-03\_05-58-36\\com.apkognito.WalkaboutMiniGolf.apk: No such file or directory
[logpack.zip](https://github.com/user-attachments/files/17997773/logpack.zip)

**Comment by Iorkca** [comment]
Hello Sombody101

This 100% worked, game is installed and functioning, really appreciate you taking the time to sort this out.  Is there something I could have changed on my end to eliminate the whitespaces, I had my output directory on an external hard driver would it have been better on an internal one?

**Comment by Sombody101** [comment]
The whitespace error wasn't due to something you did, it was me. I never tested the paths using a file path that contained a space, which is why it caused ADB to not work.

Take this command as an example:
```powershell
adb install C:\path\to\your\file.apk
```
This command works because the file argument ADB gets looks like `C:\path\to\your\file`. This changes when given this input:
```powershell
adb install C:\path\to\your\cool file.apk
```
The path is technically correct, but spaces are used to separate arguments, so ADB actually sees the path as `file.apk` and thinks `C:\path\to\your\cool` is a separate argument.

Adding quotations fixes this:
```powershell
adb install "C:\path\to\your\cool file.apk"
```
By doing that, ADB gets the full path `C:\path\to\your\cool file.apk`.

So, again, this wasn't caused by something you did, it was caused by something I failed to do (which was to test the feature correctly).

**Comment by Iorkca** [comment]
Hey again, so not sure if this is something with the renaming or not.  The game is on the headset and loads, I made the mistake of not actually trying a course the other day and none load, soon as I try to launch the course it starts to load crashes and says I don't have the content I'm trying to launch and its a course that comes with the base game.  Any thought?

**Comment by Sombody101** [comment]
So, turns out this specific game uses absolute paths to find the scenes.
Running this barbaric command proved it:
```bash
cat -v *.obb | grep "$COMPANY_NAME"
```

This complicates things because an OBB is like a ZIP (or an APK) but for Unity assets. Except _it's very large_ (509MB compressed) and can use upwards of a gigabyte of storage when fully extracted. Pretty much all OBB files are like this. So, in order to fully fix this issue, this file would need to be extracted on your drive much like the APK, renamed, and then repacked. This will take significantly more space, time, and processing power.

I've narrowed the error down to the file `/assets/aa/catalog.json` within the OBB, which lists all default scenes and maps.
![image](https://github.com/user-attachments/assets/74422334-b4c5-4016-ba25-89b17086f813)

I'm going to see if there are any libraries that can edit this file type without having to fully extract it to a user's drive.
I'll check back in later with the results.

**Comment by Electrohx** [comment]
Hey! Was about to look into this issue but the main branch is not building. I'm having a lot of issues with files missing like `LoggableObservableObject` and `AdbDeviceInfoBox`. Can you look into that and make the main branch build in VS?

**Comment by Electrohx** [comment]
Also @Sombody101 happy xmas :)

**Comment by Sombody101** [comment]
`LoggableObservableObject` is located under `APKognito.Utilities.MVVM`. I recently moved it into its own folder, so there's a chance your IDE is trying to compensate for that as the namespace for the class is still `APKognito.Utilities`.

For all files with this file missing error, ensure this line exists at the top:
```cs
using APKognito.Utilities
```

And I'm not sure what `AdbDeviceInfoBox` is. The item with the closest name to that would be `AdbDeviceInfo` (under `APKognito.Controls`), which is a control used for displaying the selected device on the view.
Is that the name given to you by the compiler error?

Either way, I've gone ahead and created a push with the entire APKognito source.
Let me know if you still run into errors

**Comment by Electrohx** [comment]
Perfect, thanks alot mate! It's building for me now. I'll let you know if I find anything solution for this issue

**Comment by Sombody101** [comment]
Well, I've technically already found the solution, but I will not be implementing it.

It's possible to fix this by using the same method used for renaming the APKs, but the overhead needed for renaming is not a good enough tradeoff for only one application.

If I find that more apps are set up like this, then I'll implement something.

**Comment by Electrohx** [comment]
Ah okey! What about implement that solution and add a checkbox for it that it will do the process only if the box is checked?

**Comment by Electrohx** [comment]
Or if you already have a version for that working, mind pushing it into a new branch?

**Comment by Sombody101** [comment]
As said, I'm not implementing it unless enough apps require it.

I'm trying to develop APKognito to use a standard process for all apps. I don't want to provide support for a process that will only for one app.

**Comment by Scrabsi** [comment]
I don't have anything particularly useful to add other than I also needed this for this same apk, as it's a Quest+ APK and those are catching errors. I respect not adding the functionality though as it does seem to be a one off issue.

**Comment by Sombody101** [comment]
I'm closing this issue as there won't be any fix for this game.

**Comment by Sombody101** [comment]
I'm reopening this because I may have inadvertently fixed the app in question.
Try using the latest public debug build and let me know how it goes.

You too, @Scrabsi.

[d1.9.9223.26483](https://github.com/Sombody101/APKognito/releases/tag/d1.9.9223.26483)

**Comment by Electrohx** [comment]
@Sombody101 Yes, I can verify that it in fact works great now. Very good job mate 👏
